prow/cmd/checkconfig/main.go

```diff
@@ -212,6 +235,89 @@ func validateURLs(c config.ProwConfig) error {
 	return errorutil.NewAggregate(validationErrs...)
 }
 
+func validateUnknownFields(cfg interface{}, cfgBytes []byte, filePath string) error {
+	var obj interface{}
+	err := yaml.Unmarshal(cfgBytes, &obj)
+	if err != nil {
+		return fmt.Errorf("error while unmarshaling yaml: %v", err)
+	}
+	unknownFields := checkUnknownFields("", obj, reflect.ValueOf(cfg))
+	if len(unknownFields) > 0 {
+		return fmt.Errorf("unknown fields present in %s: %v", filePath, strings.Join(unknownFields, ", "))
+	}
+	return nil
+}
+
+func checkUnknownFields(keyPref string, obj interface{}, cfg reflect.Value) []string {
+	var uf []string
+	switch concreteVal := obj.(type) {
+	case map[string]interface{}:
+		// Iterate over map and check every value
+		for key, val := range concreteVal {
+			fullKey := fmt.Sprintf("%s.%s", keyPref, key)
+			subCfg := getSubCfg(key, cfg)
+			if !subCfg.IsValid() {
+				uf = append(uf, fullKey[1:])
```

Comment:
Why does this remove the first character from `fullKey`?

Reply:
`fullKey` is [formed](https://github.com/kubernetes/test-infra/pull/11622/files#diff-bb97e06c2a17f654b1eab01a05d73efeR259) by appending `.<current key>` to existing path. Since in the beginning existing path is empty `fullKey`'s first character would be `.`: `.triggers[1].repoz`.

Added comment